test/test_multi_configuration.rb

```diff
@@ -0,0 +1,113 @@
+require 'al-test-utils'
+
+class TestMultiConfiguration < Test::Unit::TestCase
+  include AlTestUtils
+
+  def setup
+    super
+    @original_configs = ActiveLdap::Base.configurations.dup
+  end
+
+  def teardown
+    ActiveLdap::Base.clear_active_connections!
+    ActiveLdap::Base.configurations = @original_configs
+    super
+  end
+
+  def test_configuration_with_no_or_primary_key
+    assert_nothing_raised do
+      ActiveLdap::Base.setup_connection
+    end
+    ActiveLdap::Base.configurations[LDAP_ENV] = {
+      "primary" => @original_configs[LDAP_ENV]
+    }
+    assert_nothing_raised do
+      ActiveLdap::Base.setup_connection
+    end
```

Comment:
Can we call a method that uses the new connection instead of `assert_nothing_raised` something like the following?

```suggestion
    ActiveLdap::Base.setup_connection
    assert do
      ActiveLdap::Base.connection.connected?
    end
```

Reply:
I replaced all occurences with calling `connect`, which calls `.connection.connect`